src/Code.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { withTheme } from 'styled-components'
+import SyntaxHighlighter, { registerLanguage } from "react-syntax-highlighter/prism-light"
+import defaultTheme from 'react-syntax-highlighter/styles/prism/atom-dark'
+import javascript from 'react-syntax-highlighter/languages/prism/javascript'
+import elixir from 'react-syntax-highlighter/languages/prism/elixir'
+
+registerLanguage('javascript', javascript);
+registerLanguage('elixir', elixir);
+
+export default withTheme(class Code extends React.Component {
+  static propTypes = {
+    children: PropTypes.string,
+    className: PropTypes.string,
+    theme: PropTypes.object
+  }
+
+  getLangauge = (lang) => {
+    return lang ? lang.replace('language-', '') : 'javascript'
+  }
+
+  render() {
+    const { className, children, theme } = this.props
+    const language = this.getLangauge(className)
+    const style = (theme && theme.highlight) ? theme.highlight : defaultTheme
```

Comment:
I think `theme` will always be an object, even if it's an empty object

Reply:
I think one of the tests failed before hand, that's why I changed this to a more defensive approach. We will also need to update the tests.

Reply:
Yeah, some of the theming context stuff will fail if the provider or a default prop isn't there, not sure if that's related or not

Reply:
I think the theme context is undefined initially, but will recheck.